orchestrator-salt/src/main/resources/salt/salt/postgresql/disaster_recovery/scripts/get_backup_data_sizes.sh

```diff
@@ -0,0 +1,148 @@
+  doLog "Getting the data sizes for the local databases."
+
+  DATABASES='"hive" "ranger" "profiler_agent" "profiler_metric"'
+  RESULT=""
+
+  CUR_DB_SIZE=0
+  TRIMMED_DB=""
+  for DB in $DATABASES; do
+    TRIMMED_DB=$(echo "$DB" | tr -d '"')
+    if doesDatabaseExist "$TRIMMED_DB" ; then
+      CUR_DB_SIZE=$(runPSQLCommand "SELECT pg_database_size('${TRIMMED_DB}');")
+      RESULT="${RESULT}${DB}:${CUR_DB_SIZE},"
+      doLog "Size of database ${DB} is ${CUR_DB_SIZE} bytes."
+    fi
+  done
+
+  doLog "Finished getting the data sizes for the local databases."
+  echo "{${RESULT%?}}"
+}
+
+kinit_as() {
+  doLog "Attempting kinit as $1 using Keytab: $2"
+  kinit -kt "$2" "$1/$(hostname -f)"
+  if [[ $? -ne 0 ]]; then
+    errorExit "Couldn't kinit as $1."
+  fi
+}
+
+getDataSizesForHBase() {
+    doLog "Getting data sizes for HBase tables."
+
+    HBASE_KEYTAB=$(find /run/cloudera-scm-agent/process/ \
+      -name "*.keytab" -a \( -path "*hbase-REGIONSERVER*" -o -path "*hbase-MASTER*" \) \
+      -a -type f | head -n 1)
+    kinit_as "hbase" "$HBASE_KEYTAB"
+
+    # Get sizes of each table along with the name of the table. (Sizes are duplicated which is why we skip the second field)
+    SIZES=($(hdfs --loglevel ERROR  dfs -du -x /hbase/data/default/ 2> >(doLog) | awk '{print $1" "$3}'))
+
+    # Construct final output from response.
+    RESULT=""
+    TABLE_NAME=""
+    CUR_SIZE=""
+    IDX=0
+    {% raw %}
+    while [[ ${IDX} -lt ${#SIZES[@]} ]]; do
+    {% endraw %}
+      CUR_SIZE="${SIZES[$IDX]}"
+      TABLE_NAME="${SIZES[$((IDX+1))]##*/}" # Remove preceding directories in path to table.
+      TABLE_NAME=$(echo ${TABLE_NAME} | awk '{print tolower($0)}')
+      doLog "Size of HBase table \"${TABLE_NAME}\" is ${CUR_SIZE} bytes."
+
+      RESULT="${RESULT}\"${TABLE_NAME}\":${CUR_SIZE},"
+      IDX=$((IDX+2))
+    done
+
+    doLog "Finished getting data sizes for HBase tables."
+    echo "{${RESULT%?}}"
+}
+
+getDataSizesForSolr() {
+  doLog "Getting data sizes for Solr collections."
+
+  SOLR_KEYTAB=$(find /run/cloudera-scm-agent/process/ -name "*.keytab" -path "*solr-SOLR_SERVER*" | head -n 1)
+  kinit_as "solr" "$SOLR_KEYTAB"
+
+  # Obtain sizes for all Solr shards. Below query simply obtains the sizes of each shard for each collection, ensuring to ignore replicas by taking
+  # the maximum size replica, and then summing the sizes of the shards for each collection.
+  SIZES=$(curl -k -s --negotiate -u : "https://$(hostname -f):8985/solr/admin/cores?action=STATUS&wt=json&indent=true" | \
+    jq -c 'reduce .status[] as $x (null; .[$x.cloud.collection][$x.cloud.shard] = ([.[$x.cloud.collection][$x.cloud.shard], $x.index.sizeInBytes] | max))
+    | reduce to_entries[] as $x (null; .[$x.key] += ([$x.value[]] | add))')
+  doLog "Solr collection sizes result: ${SIZES}"
+
+  doLog "Finished getting data sizes for Solr collections."
+  echo "$SIZES"
+}
+
+outputDataSizes() {
+  printf '{"database":%s,"hbase":%s,"solr":%s}\n' "$1" "$2" "$3"
```

Comment:
Just a question, how do you process this output (which goes to STDOUT)?

Reply:
That is located in the other PR here: https://github.com/hortonworks/cloudbreak/pull/13550/files#diff-2c9c4018786eb4a3e48596a80f780879ae2b2a562abee3b958c01f6cbf4594e2 on line 85. Specifically, I make a synchronous call to Salt which returns the response within a large JSON object.